#Preparation

In [536]:
pd.set_option("display.precision", 5)

In [537]:
import pandas as pd
import numpy as np
import datetime as datetime
df = pd.read_csv('https://github.com/FTDS-learning-materials/phase-0/raw/main/src/data_ngc.csv')
df

province_state country_region        date  latitude  longitude  \
0                NaN       Malaysia  2020-03-08         2        112   
1                NaN       Malaysia  2020-03-10         2        112   
2                NaN      Singapore  2020-03-13         1        104   
3                NaN       Malaysia  2020-03-13         4        102   
4                NaN       Malaysia  2020-03-15         4        102   
...              ...            ...         ...       ...        ...   
11925         Pahang       Malaysia  2022-06-02         4        103   
11926         Pahang       Malaysia  2022-04-29         4        103   
11927         Pahang       Malaysia  2022-01-10         4        103   
11928         Pahang       Malaysia  2023-02-25         4        103   
11929         Pahang       Malaysia  2022-09-29         4        103   

                location_geom  confirmed  deaths  recovered  active  fips  \
0            POINT(112.5 2.5)         99       0         24     NaN   NaN   
1            POINT(112.5 2.5)        129       0         24     NaN   NaN   
2      POINT(103.8198 1.3521)        200       0         97     NaN   NaN   
3      POINT(101.9758 4.2105)        197       0         26     NaN   NaN   
4      POINT(101.9758 4.2105)        428       0         42     NaN   NaN   
...                       ...        ...     ...        ...     ...   ...   
11925  POINT(103.3256 3.8126)    171,734     975        NaN     NaN   NaN   
11926  POINT(103.3256 3.8126)    170,740     970        NaN     NaN   NaN   
11927  POINT(103.3256 3.8126)     97,320     782        NaN     NaN   NaN   
11928  POINT(103.3256 3.8126)    182,730   1,031        NaN     NaN   NaN   
11929  POINT(103.3256 3.8126)    177,522   1,008        NaN     NaN   NaN   

       admin2      combined_key  
0         NaN               NaN  
1         NaN               NaN  
2         NaN               NaN  
3         NaN               NaN  
4         NaN               NaN  
...       ...               ...  
11925     NaN  Pahang, Malaysia  
11926     NaN  Pahang, Malaysia  
11927     NaN  Pahang, Malaysia  
11928     NaN  Pahang, Malaysia  
11929     NaN  Pahang, Malaysia  

[11930 rows x 13 columns]

## Bikin copy agar tidak mengganggu data asli

In [538]:
df2 = df.copy()

## Cek bentuk, nama kolom, tipe data, dan jumlah nilai null tiap kolom

In [539]:
df2.shape

(11930, 13)

In [540]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11930 entries, 0 to 11929
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   province_state  9078 non-null   object 
 1   country_region  11930 non-null  object 
 2   date            11930 non-null  object 
 3   latitude        11321 non-null  float64
 4   longitude       11321 non-null  float64
 5   location_geom   11321 non-null  object 
 6   confirmed       11929 non-null  float64
 7   deaths          11913 non-null  float64
 8   recovered       1640 non-null   float64
 9   active          1520 non-null   float64
 10  fips            0 non-null      float64
 11  admin2          0 non-null      float64
 12  combined_key    11793 non-null  object 
dtypes: float64(8), object(5)
memory usage: 1.2+ MB


In [541]:
df2.isna().sum()

province_state     2852
country_region        0
date                  0
latitude            609
longitude           609
location_geom       609
confirmed             1
deaths               17
recovered         10290
active            10410
fips              11930
admin2            11930
combined_key        137
dtype: int64

#Background

By looking at the columns in the dataset, i assume this table is used to document disease or pandemic cases on some countries in the data.

The table shows us the number of confirmed, deaths, and recovered cases in a pandemic, with the location of country, region, coordinates to track the origin of the data, and also the date of data collection.

After more exploration, this table only shows us number of cases in three countries: Malaysia, Singapore, and Indonesia.

Based on that, we want to see the statistics of disease cases in those 3 countries.

#Answer

##Ganti nama beberapa kolom

- Ganti nama country_region menjadi country saja, karena di kolom tersebut memang cuma ada country.

- Ganti nama location_geom jadi coordinates, karena itu merupakan gabungan dari latitude dan longitude.

In [542]:
df2 = df2.rename(columns = {'country_region': 'country',
                            'location_geom': 'coordinates'})
df2

province_state    country        date  latitude  longitude  \
0                NaN   Malaysia  2020-03-08         2        112   
1                NaN   Malaysia  2020-03-10         2        112   
2                NaN  Singapore  2020-03-13         1        104   
3                NaN   Malaysia  2020-03-13         4        102   
4                NaN   Malaysia  2020-03-15         4        102   
...              ...        ...         ...       ...        ...   
11925         Pahang   Malaysia  2022-06-02         4        103   
11926         Pahang   Malaysia  2022-04-29         4        103   
11927         Pahang   Malaysia  2022-01-10         4        103   
11928         Pahang   Malaysia  2023-02-25         4        103   
11929         Pahang   Malaysia  2022-09-29         4        103   

                  coordinates  confirmed  deaths  recovered  active  fips  \
0            POINT(112.5 2.5)         99       0         24     NaN   NaN   
1            POINT(112.5 2.5)        129       0         24     NaN   NaN   
2      POINT(103.8198 1.3521)        200       0         97     NaN   NaN   
3      POINT(101.9758 4.2105)        197       0         26     NaN   NaN   
4      POINT(101.9758 4.2105)        428       0         42     NaN   NaN   
...                       ...        ...     ...        ...     ...   ...   
11925  POINT(103.3256 3.8126)    171,734     975        NaN     NaN   NaN   
11926  POINT(103.3256 3.8126)    170,740     970        NaN     NaN   NaN   
11927  POINT(103.3256 3.8126)     97,320     782        NaN     NaN   NaN   
11928  POINT(103.3256 3.8126)    182,730   1,031        NaN     NaN   NaN   
11929  POINT(103.3256 3.8126)    177,522   1,008        NaN     NaN   NaN   

       admin2      combined_key  
0         NaN               NaN  
1         NaN               NaN  
2         NaN               NaN  
3         NaN               NaN  
4         NaN               NaN  
...       ...               ...  
11925     NaN  Pahang, Malaysia  
11926     NaN  Pahang, Malaysia  
11927     NaN  Pahang, Malaysia  
11928     NaN  Pahang, Malaysia  
11929     NaN  Pahang, Malaysia  

[11930 rows x 13 columns]

##Ganti value yang ada di kolom coordinate


In [543]:
df2.drop(columns = 'coordinates', axis =1, inplace=True)
df2

province_state    country        date  latitude  longitude  confirmed  \
0                NaN   Malaysia  2020-03-08         2        112         99   
1                NaN   Malaysia  2020-03-10         2        112        129   
2                NaN  Singapore  2020-03-13         1        104        200   
3                NaN   Malaysia  2020-03-13         4        102        197   
4                NaN   Malaysia  2020-03-15         4        102        428   
...              ...        ...         ...       ...        ...        ...   
11925         Pahang   Malaysia  2022-06-02         4        103    171,734   
11926         Pahang   Malaysia  2022-04-29         4        103    170,740   
11927         Pahang   Malaysia  2022-01-10         4        103     97,320   
11928         Pahang   Malaysia  2023-02-25         4        103    182,730   
11929         Pahang   Malaysia  2022-09-29         4        103    177,522   

       deaths  recovered  active  fips  admin2      combined_key  
0           0         24     NaN   NaN     NaN               NaN  
1           0         24     NaN   NaN     NaN               NaN  
2           0         97     NaN   NaN     NaN               NaN  
3           0         26     NaN   NaN     NaN               NaN  
4           0         42     NaN   NaN     NaN               NaN  
...       ...        ...     ...   ...     ...               ...  
11925     975        NaN     NaN   NaN     NaN  Pahang, Malaysia  
11926     970        NaN     NaN   NaN     NaN  Pahang, Malaysia  
11927     782        NaN     NaN   NaN     NaN  Pahang, Malaysia  
11928   1,031        NaN     NaN   NaN     NaN  Pahang, Malaysia  
11929   1,008        NaN     NaN   NaN     NaN  Pahang, Malaysia  

[11930 rows x 12 columns]

In [544]:
df2.insert(5, 'coordinates', df2['latitude'].astype(str) + ',' + df2['longitude'].astype(str))
df2

province_state    country        date  latitude  longitude  \
0                NaN   Malaysia  2020-03-08         2        112   
1                NaN   Malaysia  2020-03-10         2        112   
2                NaN  Singapore  2020-03-13         1        104   
3                NaN   Malaysia  2020-03-13         4        102   
4                NaN   Malaysia  2020-03-15         4        102   
...              ...        ...         ...       ...        ...   
11925         Pahang   Malaysia  2022-06-02         4        103   
11926         Pahang   Malaysia  2022-04-29         4        103   
11927         Pahang   Malaysia  2022-01-10         4        103   
11928         Pahang   Malaysia  2023-02-25         4        103   
11929         Pahang   Malaysia  2022-09-29         4        103   

           coordinates  confirmed  deaths  recovered  active  fips  admin2  \
0            2.5,112.5         99       0         24     NaN   NaN     NaN   
1            2.5,112.5        129       0         24     NaN   NaN     NaN   
2      1.3521,103.8198        200       0         97     NaN   NaN     NaN   
3      4.2105,101.9758        197       0         26     NaN   NaN     NaN   
4      4.2105,101.9758        428       0         42     NaN   NaN     NaN   
...                ...        ...     ...        ...     ...   ...     ...   
11925  3.8126,103.3256    171,734     975        NaN     NaN   NaN     NaN   
11926  3.8126,103.3256    170,740     970        NaN     NaN   NaN     NaN   
11927  3.8126,103.3256     97,320     782        NaN     NaN   NaN     NaN   
11928  3.8126,103.3256    182,730   1,031        NaN     NaN   NaN     NaN   
11929  3.8126,103.3256    177,522   1,008        NaN     NaN   NaN     NaN   

           combined_key  
0                   NaN  
1                   NaN  
2                   NaN  
3                   NaN  
4                   NaN  
...                 ...  
11925  Pahang, Malaysia  
11926  Pahang, Malaysia  
11927  Pahang, Malaysia  
11928  Pahang, Malaysia  
11929  Pahang, Malaysia  

[11930 rows x 13 columns]

## Ganti tipe data

In [545]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11930 entries, 0 to 11929
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   province_state  9078 non-null   object 
 1   country         11930 non-null  object 
 2   date            11930 non-null  object 
 3   latitude        11321 non-null  float64
 4   longitude       11321 non-null  float64
 5   coordinates     11930 non-null  object 
 6   confirmed       11929 non-null  float64
 7   deaths          11913 non-null  float64
 8   recovered       1640 non-null   float64
 9   active          1520 non-null   float64
 10  fips            0 non-null      float64
 11  admin2          0 non-null      float64
 12  combined_key    11793 non-null  object 
dtypes: float64(8), object(5)
memory usage: 1.2+ MB


In [546]:
df2['date'] = pd.to_datetime(df2['date'])
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11930 entries, 0 to 11929
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   province_state  9078 non-null   object        
 1   country         11930 non-null  object        
 2   date            11930 non-null  datetime64[ns]
 3   latitude        11321 non-null  float64       
 4   longitude       11321 non-null  float64       
 5   coordinates     11930 non-null  object        
 6   confirmed       11929 non-null  float64       
 7   deaths          11913 non-null  float64       
 8   recovered       1640 non-null   float64       
 9   active          1520 non-null   float64       
 10  fips            0 non-null      float64       
 11  admin2          0 non-null      float64       
 12  combined_key    11793 non-null  object        
dtypes: datetime64[ns](1), float64(8), object(4)
memory usage: 1.2+ MB


## Drop beberapa kolom

Drop kolom yang null valuenya sampai 80% lebih dari data

In [547]:
df2.drop(columns = ['fips', 'admin2'], axis =1, inplace=True)
df2

province_state    country       date  latitude  longitude  \
0                NaN   Malaysia 2020-03-08         2        112   
1                NaN   Malaysia 2020-03-10         2        112   
2                NaN  Singapore 2020-03-13         1        104   
3                NaN   Malaysia 2020-03-13         4        102   
4                NaN   Malaysia 2020-03-15         4        102   
...              ...        ...        ...       ...        ...   
11925         Pahang   Malaysia 2022-06-02         4        103   
11926         Pahang   Malaysia 2022-04-29         4        103   
11927         Pahang   Malaysia 2022-01-10         4        103   
11928         Pahang   Malaysia 2023-02-25         4        103   
11929         Pahang   Malaysia 2022-09-29         4        103   

           coordinates  confirmed  deaths  recovered  active      combined_key  
0            2.5,112.5         99       0         24     NaN               NaN  
1            2.5,112.5        129       0         24     NaN               NaN  
2      1.3521,103.8198        200       0         97     NaN               NaN  
3      4.2105,101.9758        197       0         26     NaN               NaN  
4      4.2105,101.9758        428       0         42     NaN               NaN  
...                ...        ...     ...        ...     ...               ...  
11925  3.8126,103.3256    171,734     975        NaN     NaN  Pahang, Malaysia  
11926  3.8126,103.3256    170,740     970        NaN     NaN  Pahang, Malaysia  
11927  3.8126,103.3256     97,320     782        NaN     NaN  Pahang, Malaysia  
11928  3.8126,103.3256    182,730   1,031        NaN     NaN  Pahang, Malaysia  
11929  3.8126,103.3256    177,522   1,008        NaN     NaN  Pahang, Malaysia  

[11930 rows x 11 columns]

Drop kolom province_state dan combined_key karena yang punya nilai itu hanya negara Malaysia, sedangkan Indonesia dan Singapura tidak punya. Sulit untuk mengisi provinsi untuk kedua negara tersebut. combined_key sendiri adalah kode yang menggabungkan provinsi dengan negaranya.

In [548]:
df2.drop(columns = ['province_state','combined_key'], axis=1, inplace=True)
df2

country       date  latitude  longitude      coordinates  confirmed  \
0       Malaysia 2020-03-08         2        112        2.5,112.5         99   
1       Malaysia 2020-03-10         2        112        2.5,112.5        129   
2      Singapore 2020-03-13         1        104  1.3521,103.8198        200   
3       Malaysia 2020-03-13         4        102  4.2105,101.9758        197   
4       Malaysia 2020-03-15         4        102  4.2105,101.9758        428   
...          ...        ...       ...        ...              ...        ...   
11925   Malaysia 2022-06-02         4        103  3.8126,103.3256    171,734   
11926   Malaysia 2022-04-29         4        103  3.8126,103.3256    170,740   
11927   Malaysia 2022-01-10         4        103  3.8126,103.3256     97,320   
11928   Malaysia 2023-02-25         4        103  3.8126,103.3256    182,730   
11929   Malaysia 2022-09-29         4        103  3.8126,103.3256    177,522   

       deaths  recovered  active  
0           0         24     NaN  
1           0         24     NaN  
2           0         97     NaN  
3           0         26     NaN  
4           0         42     NaN  
...       ...        ...     ...  
11925     975        NaN     NaN  
11926     970        NaN     NaN  
11927     782        NaN     NaN  
11928   1,031        NaN     NaN  
11929   1,008        NaN     NaN  

[11930 rows x 9 columns]

In [549]:
df2.isna().sum()

country            0
date               0
latitude         609
longitude        609
coordinates        0
confirmed          1
deaths            17
recovered      10290
active         10410
dtype: int64

In [550]:
df2.drop(columns = ['recovered','active'], axis=1, inplace=True)
df2

country       date  latitude  longitude      coordinates  confirmed  \
0       Malaysia 2020-03-08         2        112        2.5,112.5         99   
1       Malaysia 2020-03-10         2        112        2.5,112.5        129   
2      Singapore 2020-03-13         1        104  1.3521,103.8198        200   
3       Malaysia 2020-03-13         4        102  4.2105,101.9758        197   
4       Malaysia 2020-03-15         4        102  4.2105,101.9758        428   
...          ...        ...       ...        ...              ...        ...   
11925   Malaysia 2022-06-02         4        103  3.8126,103.3256    171,734   
11926   Malaysia 2022-04-29         4        103  3.8126,103.3256    170,740   
11927   Malaysia 2022-01-10         4        103  3.8126,103.3256     97,320   
11928   Malaysia 2023-02-25         4        103  3.8126,103.3256    182,730   
11929   Malaysia 2022-09-29         4        103  3.8126,103.3256    177,522   

       deaths  
0           0  
1           0  
2           0  
3           0  
4           0  
...       ...  
11925     975  
11926     970  
11927     782  
11928   1,031  
11929   1,008  

[11930 rows x 7 columns]

## Drop baris yang kosong

In [551]:
df2.isna().sum()

country          0
date             0
latitude       609
longitude      609
coordinates      0
confirmed        1
deaths          17
dtype: int64

In [552]:
df2.dropna(subset=['latitude'], inplace=True)
df2

country       date  latitude  longitude      coordinates  confirmed  \
0       Malaysia 2020-03-08         2        112        2.5,112.5         99   
1       Malaysia 2020-03-10         2        112        2.5,112.5        129   
2      Singapore 2020-03-13         1        104  1.3521,103.8198        200   
3       Malaysia 2020-03-13         4        102  4.2105,101.9758        197   
4       Malaysia 2020-03-15         4        102  4.2105,101.9758        428   
...          ...        ...       ...        ...              ...        ...   
11925   Malaysia 2022-06-02         4        103  3.8126,103.3256    171,734   
11926   Malaysia 2022-04-29         4        103  3.8126,103.3256    170,740   
11927   Malaysia 2022-01-10         4        103  3.8126,103.3256     97,320   
11928   Malaysia 2023-02-25         4        103  3.8126,103.3256    182,730   
11929   Malaysia 2022-09-29         4        103  3.8126,103.3256    177,522   

       deaths  
0           0  
1           0  
2           0  
3           0  
4           0  
...       ...  
11925     975  
11926     970  
11927     782  
11928   1,031  
11929   1,008  

[11321 rows x 7 columns]

In [553]:
df2.isna().sum()

country        0
date           0
latitude       0
longitude      0
coordinates    0
confirmed      0
deaths         0
dtype: int64

In [554]:
df2

country       date  latitude  longitude      coordinates  confirmed  \
0       Malaysia 2020-03-08         2        112        2.5,112.5         99   
1       Malaysia 2020-03-10         2        112        2.5,112.5        129   
2      Singapore 2020-03-13         1        104  1.3521,103.8198        200   
3       Malaysia 2020-03-13         4        102  4.2105,101.9758        197   
4       Malaysia 2020-03-15         4        102  4.2105,101.9758        428   
...          ...        ...       ...        ...              ...        ...   
11925   Malaysia 2022-06-02         4        103  3.8126,103.3256    171,734   
11926   Malaysia 2022-04-29         4        103  3.8126,103.3256    170,740   
11927   Malaysia 2022-01-10         4        103  3.8126,103.3256     97,320   
11928   Malaysia 2023-02-25         4        103  3.8126,103.3256    182,730   
11929   Malaysia 2022-09-29         4        103  3.8126,103.3256    177,522   

       deaths  
0           0  
1           0  
2           0  
3           0  
4           0  
...       ...  
11925     975  
11926     970  
11927     782  
11928   1,031  
11929   1,008  

[11321 rows x 7 columns]

In [555]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11321 entries, 0 to 11929
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   country      11321 non-null  object        
 1   date         11321 non-null  datetime64[ns]
 2   latitude     11321 non-null  float64       
 3   longitude    11321 non-null  float64       
 4   coordinates  11321 non-null  object        
 5   confirmed    11321 non-null  float64       
 6   deaths       11321 non-null  float64       
dtypes: datetime64[ns](1), float64(4), object(2)
memory usage: 707.6+ KB


## Ganti tipe data [confirmed] dan [deaths] ke **integer**

In [556]:
df2['confirmed'] = df2['confirmed'].astype(int)
df2['deaths'] = df2['deaths'].astype(int)

In [557]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11321 entries, 0 to 11929
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   country      11321 non-null  object        
 1   date         11321 non-null  datetime64[ns]
 2   latitude     11321 non-null  float64       
 3   longitude    11321 non-null  float64       
 4   coordinates  11321 non-null  object        
 5   confirmed    11321 non-null  int64         
 6   deaths       11321 non-null  int64         
dtypes: datetime64[ns](1), float64(2), int64(2), object(2)
memory usage: 707.6+ KB


##Cleaned (?) Data

In [558]:
df2.head(20)

country       date  latitude  longitude      coordinates  confirmed  \
0    Malaysia 2020-03-08         2        112        2.5,112.5         99   
1    Malaysia 2020-03-10         2        112        2.5,112.5        129   
2   Singapore 2020-03-13         1        104  1.3521,103.8198        200   
3    Malaysia 2020-03-13         4        102  4.2105,101.9758        197   
4    Malaysia 2020-03-15         4        102  4.2105,101.9758        428   
5   Singapore 2020-03-15         1        104  1.3521,103.8198        226   
6    Malaysia 2020-03-02         2        112        2.5,112.5         29   
7    Malaysia 2020-03-11         2        112        2.5,112.5        149   
8    Malaysia 2020-03-14         4        102  4.2105,101.9758        238   
9   Singapore 2020-03-14         1        104  1.3521,103.8198        212   
10   Malaysia 2020-03-19         4        102  4.2105,101.9758        900   
11  Singapore 2020-03-19         1        104  1.3521,103.8198        345   
12   Malaysia 2020-03-03         2        112        2.5,112.5         36   
13   Malaysia 2020-03-04         2        112        2.5,112.5         50   
14   Malaysia 2020-03-09         2        112        2.5,112.5        117   
15   Malaysia 2020-03-20         4        102  4.2105,101.9758       1030   
16  Singapore 2020-03-20         1        104  1.3521,103.8198        385   
17   Malaysia 2020-03-12         2        112        2.5,112.5        149   
18   Malaysia 2020-03-06         2        112        2.5,112.5         83   
19   Malaysia 2020-03-16         4        102  4.2105,101.9758        566   

    deaths  
0        0  
1        0  
2        0  
3        0  
4        0  
5        0  
6        0  
7        0  
8        0  
9        0  
10       2  
11       0  
12       0  
13       0  
14       0  
15       3  
16       0  
17       0  
18       0  
19       0

In [559]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 11321 entries, 0 to 11929
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   country      11321 non-null  object        
 1   date         11321 non-null  datetime64[ns]
 2   latitude     11321 non-null  float64       
 3   longitude    11321 non-null  float64       
 4   coordinates  11321 non-null  object        
 5   confirmed    11321 non-null  int64         
 6   deaths       11321 non-null  int64         
dtypes: datetime64[ns](1), float64(2), int64(2), object(2)
memory usage: 707.6+ KB


## Descriptive Analytics

In [560]:
df2['country'].unique()

array(['Malaysia', 'Singapore', 'Indonesia'], dtype=object)

In [561]:
malaysia = df2[df2['country'] =='Malaysia']

In [562]:
malaysia_sorted = malaysia.sort_values(by='date')
malaysia_sorted

country       date  latitude  longitude      coordinates  confirmed  \
29     Malaysia 2020-03-01         2        112        2.5,112.5         29   
6      Malaysia 2020-03-02         2        112        2.5,112.5         29   
12     Malaysia 2020-03-03         2        112        2.5,112.5         36   
13     Malaysia 2020-03-04         2        112        2.5,112.5         50   
28     Malaysia 2020-03-05         2        112        2.5,112.5         50   
...         ...        ...       ...        ...              ...        ...   
10882  Malaysia 2023-03-09         1        104  1.4854,103.7618     398660   
2276   Malaysia 2023-03-09         6        100  6.4449,100.2048      19014   
8212   Malaysia 2023-03-09         2        102  2.1896,102.2501     145426   
7144   Malaysia 2023-03-09         5        101  4.5921,101.0901     239281   
3344   Malaysia 2023-03-09         2        110  1.5533,110.3592     324659   

       deaths  
29          0  
6           0  
12          0  
13          0  
28          0  
...       ...  
10882    4715  
2276      198  
8212     1203  
7144     2125  
3344     1788  

[9114 rows x 7 columns]

In [563]:
pd.options.display.float_format = '{:,.0f}'.format

In [564]:
malaysia.describe()

latitude  longitude  confirmed  deaths
count     9,114      9,114      9,114   9,114
mean          4        104    260,498   2,371
std           2          5    307,450   8,146
min           1        100         29       0
25%           3        102     91,147     816
50%           4        102    202,075   1,462
75%           5        103    308,284   2,678
max           6        116  2,127,934 624,641

Because it's a cumulative data, we get the total number of cases by looking at the max value.

Based on the statistics, the total number of disease cases in Malaysia is 2,127,934  with confirmed deaths reaching 624,641. The rasio of confirmed and deaths cases in Malaysia seems to be off and could indicate an error in the data input process.

In [565]:
singapore = df2[df2['country'] =='Singapore']
singapore

country       date  latitude  longitude      coordinates  confirmed  \
2     Singapore 2020-03-13         1        104  1.3521,103.8198        200   
5     Singapore 2020-03-15         1        104  1.3521,103.8198        226   
9     Singapore 2020-03-14         1        104  1.3521,103.8198        212   
11    Singapore 2020-03-19         1        104  1.3521,103.8198        345   
16    Singapore 2020-03-20         1        104  1.3521,103.8198        385   
...         ...        ...       ...        ...              ...        ...   
6016  Singapore 2022-04-29         1        104  1.2833,103.8333    1193250   
6017  Singapore 2021-03-06         1        104  1.2833,103.8333      60020   
6018  Singapore 2022-01-10         1        104  1.2833,103.8333     286397   
6019  Singapore 2023-02-25         1        104  1.2833,103.8333    2222006   
6020  Singapore 2022-09-29         1        104  1.2833,103.8333    1904192   

      deaths  
2          0  
5          0  
9          0  
11         0  
16         0  
...      ...  
6016    1334  
6017      29  
6018     838  
6019    1722  
6020    1617  

[1104 rows x 7 columns]

In [566]:
singapore.describe()

latitude  longitude  confirmed  deaths
count     1,104      1,104      1,104   1,104
mean          1        104    655,273     626
std           0          0    821,333     697
min           1        104        106       0
25%           1        104     58,230      29
50%           1        104     68,340      55
75%           1        104  1,322,265   1,393
max           1        104  2,235,294   1,722

Based on the statistics, the total number of the disease cases in Singapore is 2,235,294 with confirmed deaths of 1,722.

In [567]:
indonesia = df2[df2['country'] =='Indonesia']
indonesia

country       date  latitude  longitude       coordinates  confirmed  \
6021  Indonesia 2020-05-15        -1        114  -0.7893,113.9213      16496   
6022  Indonesia 2022-05-26        -1        114  -0.7893,113.9213    6053670   
6023  Indonesia 2020-03-08        -1        114  -0.7893,113.9213          6   
6024  Indonesia 2023-02-13        -1        114  -0.7893,113.9213    6732968   
6025  Indonesia 2022-02-16        -1        114  -0.7893,113.9213    4966046   
...         ...        ...       ...        ...               ...        ...   
7119  Indonesia 2022-04-29        -1        114  -0.7893,113.9213    6046467   
7120  Indonesia 2021-03-06        -1        114  -0.7893,113.9213    1373836   
7121  Indonesia 2022-01-10        -1        114  -0.7893,113.9213    4266649   
7122  Indonesia 2023-02-25        -1        114  -0.7893,113.9213    6735451   
7123  Indonesia 2022-09-29        -1        114  -0.7893,113.9213    6429767   

      deaths  
6021    1076  
6022  156556  
6023       0  
6024  160870  
6025  145622  
...      ...  
7119  156240  
7120   37154  
7121  144136  
7122  160905  
7123  158093  

[1103 rows x 7 columns]

In [568]:
indonesia.describe()

latitude  longitude  confirmed  deaths
count     1,103      1,103      1,103   1,103
mean         -1        114  3,347,241  90,965
std           0          0  2,575,292  66,299
min          -1        114          2       0
25%          -1        114    553,692  17,277
50%          -1        114  4,123,617 135,469
75%          -1        114  6,056,017 156,604
max          -1        114  6,738,225 160,941

Based on the statistics, the total number of the disease cases in Indonesia is 6,738,225 with confirmed deaths of 160,941.

#Conclusion

Based on the statistics, the total number of disease cases in Malaysia is 2,127,934 with confirmed deaths reaching 624,641. The rasio of confirmed and deaths cases in Malaysia seems to be off and could indicate an error in the data input process or other outliers

In Singapore, the number of cases is higher than Malaysia with 2,235,294 but the number of deaths is significantly lower with 1,722 confirmed deaths.

Lastly in Indonesia, the number of cases is the highest among the SEA countries with total confirmed cases reaching 6,738,225 and death toll of 160,941. This could be caused by Indonesia having the most population in the South-East Asia region with around 278 million people, compared to Singapore with only 5.4 million. Indonesia also ranked #4 in population globally.
